pkg/application/doguInstallationUseCase.go

```diff
@@ -0,0 +1,30 @@
+package application
+
+import (
+	"fmt"
+	"github.com/cloudogu/k8s-blueprint-operator/pkg/domainservice"
+)
+
+type DoguInstallationUseCase struct {
+	doguRepo          domainservice.DoguInstallationRepository
+	doguDomainUseCase domainservice.DoguInstallationDomainUseCase
+}
+
+func (useCase *DoguInstallationUseCase) validateDoguHealth() error {
+	installedDogus, err := useCase.doguRepo.GetAll()
+	if err != nil {
+		return fmt.Errorf("cannot evaluate dogu health states: %w", err)
+	}
+
+	return useCase.doguDomainUseCase.ValidateDoguHealth(installedDogus)
+}
+
+func (useCase *DoguInstallationUseCase) installDogu(dogu string, version string) error {
+	//TODO
+	return nil
+}
+
+func (useCase *DoguInstallationUseCase) uninstallDogu(qualifiedDoguName string) error {
```

Comment:
🚫 **[checkstyle]** <unused> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>func `(*DoguInstallationUseCase).uninstallDogu` is unused

Reply:
Will be used later